### PR TITLE
address API review comments for extra mappings

### PR DESCRIPTION
**Pull request by aramase**

### CODE REVIEW

**Comment by aramase** [comment]
cc @enj

File: keps/sig-auth/3331-structured-config-for-oidc-authentication/README.md [hunk 1]
```diff
@@ -446,36 +442,38 @@ type JWTAuthenticator struct {
...
         // +optional
         Extra []ExtraMapping `json:"extra,omitempty"`
     }
 
     type ExtraMapping struct {
-        // Key is a CEL expression to extract extra attribute key.
-        // Claim must be a singular string claim.
+        // Key is a string to use as the extra attribute key.
+        Key string `json:"key"`
+        // ValueExpression is a CEL expression to extract extra attribute value.
```
Comment (enj) [inline keps/sig-auth/3331-structured-config-for-oidc-authentication/README.md]:
```suggestion
        // valueExpression is a CEL expression to extract extra attribute value.
```

File: keps/sig-auth/3331-structured-config-for-oidc-authentication/README.md [hunk 2]
```diff
@@ -446,36 +442,38 @@ type JWTAuthenticator struct {
...
         // Expression must produce a string value.
         // "" and null values are treated as the extra mapping not being present.
```
Comment (enj) [inline keps/sig-auth/3331-structured-config-for-oidc-authentication/README.md]:
```suggestion
        // "", [], and null values are treated as the extra mapping not being present.
        // Empty string values contained within a string array are filtered out.
```

File: keps/sig-auth/3331-structured-config-for-oidc-authentication/README.md [hunk 3]
```diff
@@ -446,36 +442,38 @@ type JWTAuthenticator struct {
...
         Extra []ExtraMapping `json:"extra,omitempty"`
     }
 
     type ExtraMapping struct {
-        // Key is a CEL expression to extract extra attribute key.
-        // Claim must be a singular string claim.
+        // Key is a string to use as the extra attribute key.
+        Key string `json:"key"`
+        // ValueExpression is a CEL expression to extract extra attribute value.
         // Expression must produce a string value.
```
Comment (enj) [inline keps/sig-auth/3331-structured-config-for-oidc-authentication/README.md]:
```suggestion
        // valueExpression must produce a string or string array value.
```

File: keps/sig-auth/3331-structured-config-for-oidc-authentication/README.md [hunk 4]
```diff
@@ -446,36 +442,38 @@ type JWTAuthenticator struct {
...
-        //     claim: "hd"
+        // hard-coded key, value derived from claim value
+        // - key: "admin"
+        //   valueExpression: '(has(claims.is_admin) && claims.is_admin) ? "true":""'
         //
-        // # known key, value constructed by expression
-        // - key: "example.com/myextrakey"
-        //   value:
-        //     expression: claims.someclaim+":"+claims.someclaim
+        // If multiple mappings have the same key, the result will be a concatenation of all values.
```
Comment (enj) [inline keps/sig-auth/3331-structured-config-for-oidc-authentication/README.md]:
```suggestion
        // If multiple mappings have the same key, the result will be a concatenation of all values with the ordering preserved.
```

**Pull request merged**